app/components/primer/beta/avatar.rb

```diff
@@ -65,15 +70,15 @@ def initialize(src:, alt:, size: DEFAULT_SIZE, shape: DEFAULT_SHAPE, href: nil,
           system_arguments[:classes],
           "avatar",
           "avatar-small" => size < SMALL_THRESHOLD,
-          "circle" => shape == DEFAULT_SHAPE,
-          "lh-0" => href # Addresses an overflow issue with linked avatars
+          "circle" => shape == DEFAULT_SHAPE
         )
       end
 
       def call
         if @href
-          render(Primer::LinkComponent.new(href: @href, classes: @system_arguments[:classes])) do
-            render(Primer::BaseComponent.new(**@system_arguments.except(:classes))) { content }
+          (@link_arguments[:classes] ||= []) << "lh-0" # Addresses an overflow issue with linked avatars
+          render(Primer::LinkComponent.new(href: @href, **@link_arguments)) do
+            render(Primer::BaseComponent.new(**@system_arguments)) { content }
```

Comment:
Do we want to keep `.except(:classes)` here to maintain backwards compatibility?